Spewing of "unknown pragma" warnings on MSVC

It appears one of the more recent changes cause this to be spewed all over the MSVC build (probably from [this](https://github.com/eranpeer/FakeIt/pull/214) PR). See the latest [build](https://ci.appveyor.com/project/eranpeer/fakeit/builds/38327359).

```c++
c:\projects\fakeit\include\mockutils\union_cast.hpp(12): warning C5030: attribute 'gnu::optimize' is not recognized [C:\projects\fakeit\tests\all_tests.vcxproj]
c:\projects\fakeit\include\mockutils\vtutils.hpp(25): warning C4068: unknown pragma [C:\projects\fakeit\tests\all_tests.vcxproj]
c:\projects\fakeit\include\mockutils\vtutils.hpp(26): warning C4068: unknown pragma [C:\projects\fakeit\tests\all_tests.vcxproj]
c:\projects\fakeit\include\mockutils\vtutils.hpp(33): warning C4068: unknown pragma [C:\projects\fakeit\tests\all_tests.vcxproj]
```

--- a/include/mockutils/VTUtils.hpp
+++ b/include/mockutils/VTUtils.hpp
@@ -22,15 +22,23 @@ namespace fakeit {
     class VTUtils {
     public:
 
+#ifdef __GNUG__
+#ifndef __clang__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wcast-function-type"
+#endif
+#endif
         template<typename C, typename R, typename ... arglist>
         static unsigned int getOffset(R (C::*vMethod)(arglist...)) {
             auto sMethod = reinterpret_cast<unsigned int (VirtualOffsetSelector::*)(int)>(vMethod);
             VirtualOffsetSelector offsetSelctor;
             return (offsetSelctor.*sMethod)(0);
         }
+#ifdef __GNUG__
+#ifndef __clang__
 #pragma GCC diagnostic pop
+#endif
+#endif
 
         template<typename C>
         static typename std::enable_if<std::has_virtual_destructor<C>::value, unsigned int>::type

--- a/include/mockutils/union_cast.hpp
+++ b/include/mockutils/union_cast.hpp
@@ -6,10 +6,17 @@
  * Created on Mar 10, 2014
  */
 #pragma once
+
+#if defined(__GNUG__) && !defined(__clang__)
+#define FAKEIT_NO_DEVIRTUALIZE_ATTR [[gnu::optimize("no-devirtualize")]]
+#else
+#define FAKEIT_NO_DEVIRTUALIZE_ATTR
+#endif 
+
 namespace fakeit {
 
     template<typename TARGET, typename SOURCE>
-    [[gnu::optimize("no-devirtualize")]]
+    FAKEIT_NO_DEVIRTUALIZE_ATTR
     TARGET union_cast(SOURCE source) {
         //static_assert(sizeof(TARGET) == sizeof(SOURCE), "can't convert");
         union {

